packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -827,6 +842,12 @@ class MediaQuery extends InheritedWidget {
     return MediaQuery.of(context, nullOk: true)?.boldText ?? false;
   }
 
+  /// Returns the enableOnOffSwitchLabels accessibility setting for the nearest
+  /// MediaQuery ancestor, or false if no such ancestor exists.
+  static bool onOffSwitchLabelsEnabled(BuildContext context) {
```

Comment:
I wouldn't add any new APIs to media query

Reply:
I added this comment to the original issue, probably belongs here. I agree with Jonah.

I think it would be best if we interested a new InheritedWidget in the cupertino library, rather than extending MediaQuery et al. Most of the features exposed by MediaQuery are more or less cross-platform. This "on/off labels flag" is specific to iOS and there are more flags like it. Instead of extending MediaQuery you could create a similar widget in the cupertino library called, say, CupertinoSystemSettings.

Reply:
Sounds awesome -- thanks for the detail. One question: if I go that route, would the AccessibilityFeatures interface still be best for passing the bit from the platform? (I see a similar iOS-only bit managed there: "reduceMotion")

Reply:
Yes good point: It looks like AccessibilityFeatures is the right place to plumb this value through from the engine.  WidgetsApp already rebuilds when AccessibilityFeatures  changes. That should be good enough for now, I think we can forgo creating an inherited CupertinoSystemSettings widget.